src_lockfree/bounded_stack.ml

```diff
@@ -0,0 +1,156 @@
+open Picos
+
+type 'a node = Nil | Cons of { value : 'a; tail : 'a node; capacity : int }
+
+type 'a t = {
+  head : 'a node Atomic.t;
+  cons_waiters : Trigger.t list Atomic.t;
+  prod_waiters : Trigger.t list Atomic.t;
+  capacity : int;
+}
+
+let create ?(capacity = Int.max_int) () =
+  let head = Atomic.make_contended Nil in
+  let cons_waiters = Atomic.make_contended [] in
+  let prod_waiters = Atomic.make_contended [] in
+  { head; cons_waiters; prod_waiters; capacity = max capacity 1 }
+  |> Multicore_magic.copy_as_padded
```

Comment:
Note that all the padding here takes quite a bit of memory.  A single non-padded atomic takes only 2 words.  Here we have 5 padded objects.  Each padded object takes 8 or 16 words (possibly even 32 on some architectures).  So, this allocates 40 or 80 words or 320 or 640 bytes, which is, relatively speaking, a lot.  At this point I'm mostly mentioning this as something to work on later.